lms/services/assignment.py

```diff
@@ -56,59 +71,49 @@ def get_copied_from_assignment(self, lti_params) -> Optional[Assignment]:
 
         return None
 
-    # pylint: disable=too-many-arguments
-    def upsert_assignment(
-        self,
-        tool_consumer_instance_guid,
-        resource_link_id,
-        document_url,
-        lti_params: LTIParams,
-        is_gradable=False,
-        extra=None,
-    ) -> Assignment:
-        """
-        Update or create an assignment with the given document_url.
-
-        Set the document_url for the assignment that matches
-        tool_consumer_instance_guid and resource_link_id
-        or create a new one if none exist on the DB.
+    def get_assignment_for_launch(self, request):
```

Comment:
I generally don't like very broad parameters like `request` but in this case, is a `here the request, give me an assignment`.

The rest of the methods in the servie have better scoped parameters.